### ISSUE TITLE
Duration argument being ignored in Firmin.animate

### ISSUE THREAD

**Issue by mikechambers** [beastaugh/firmin#20]
I may be doing something wrong in my code, but, I cannot get the duration property to be recognized when running Firmin.animate. Here is some example code:

```
function init()
{   
    document.onclick = onSurfaceClick;
}

function onSurfaceClick(e)
{
    console.log(e);
    var box = createBox();

    Firmin.animate(box, {
        translate:{
            x:e.pageX,
            y:e.pageY
        }
    },
    "0.7s");

}

function createBox()
{
    var box = document.createElement("div");
    box.className = "box";
    document.body.appendChild(box);
    return box;
}

window.onload = init;
```

Here is the css for the div

```
.box
{
    width:25px;
    height:25px;
    background-color:#FF0000;
    position:absolute;
    left:0px;
    top:0px;
}
```

Firmin applies the x / y translation, but the divs are not animated (they are just instantly repositioned). 

I have tried various different formats of specifying the time, but have had no luck. I can't confirm whether this is my code, the Firmin as the included Firmin tests are missing some files (I logged a separate issue on this).

Mac OS X : 10.6.7
Google Chrome : 12.0.742.91
Safari : 5.0.5 (6533.21.1)

Update : Just an update, but I have isolated the issue. The animation will not work if the div is added right before it is told to animate. You can confirm this by creating a single box in the init method, then animating it in the onclick method.

This is not specific to Firmin, but also applies when using the css transitions directly.

One solution would be for firmin to put in a slight delay when calling starting the animation. Setting the duration property in the animate api does not fix the issue.

**Comment by beastaugh** [comment]
> I have isolated the issue. The animation will not work if the div is added right before it is told to animate.

I found the same thing when trying out your test case. This is something I'll have to have a think about. It would be nice if straightforward code like yours worked without strange hacks like `setTimeout` calls, but there may be drawbacks to this approach that I haven't thought of yet.

**Comment by beastaugh** [comment]
I've pushed a fix for this in 99a13681b88dd2e31cad, using `setTimeout`. I'm going to investigate potential improvements to the mechanism such as using `window.postMessage`, as suggested by David Baron in [his article on faster timeouts](http://dbaron.org/log/20100309-faster-timeouts), but this approach will do for now.
